source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,162 @@ string as follow:
+- Set your credentials in the appropriate location 
+
+You can specify the authentication mechanism and source either
+by using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying the ``MONGODB-AWS`` authentication mechanism
+and external authentication source:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-aws-automatic-connection-string
+      
+      .. code-block:: csharp
+
+         var connectionString = "mongodb+srv://<hostname>[:<port>]?authMechanism=MONGODB-AWS&authSource=$external";
+         var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+         var client = new MongoClient(mongoClientSettings);
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-aws-automatic-mongo-credential
+
+      .. code-block:: csharp
+
+         var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<hostname>[:<port>]");
+         mongoClientSettings.Credential = new MongoCredential("MONGODB-AWS", new MongoExternalAwsIdentity(), new ExternalEvidence());
+         var client = new MongoClient(mongoClientSettings);
+   
+After you specify the authentication mechanism and source, you must set
+your credentials in the location appropriate to the credential type. The {+driver-short+} 
+checks for credentials in the following locations in the order listed here:
+
+- Web identity provider
+- Locally stored credentials
+- Environment variables
+- ECS container credentials
+- EC2 container credentials
+
+You can use an OpenID Connect (OIDC)-compatible **web identity provider** to authenticate
+to Amazon Elastic Kubernetes Service (EKS) or other services.
+To use a web identity provider, create a file that contains your
+OIDC token, then set the absolute path to this file in an environment variable by using 
+``bash`` or a similar shell as shown in the following example:
+
+.. code-block:: bash
+
+   export AWS_WEB_IDENTITY_TOKEN_FILE=<absolute path to file containing your OIDC token>
+
+To authenticate by using **locally stored credentials**, use the AWS CLI to create the 
+appropriate credential files.
+
+To retrieve credentials directly from **environment variables**, set the following 
+environment variables by using ``bash`` or a similar shell:
+
+.. code-block:: bash
+
+   export AWS_ACCESS_KEY_ID=<awsKeyId>
+   export AWS_SECRET_ACCESS_KEY=<awsSecretKey>
+   export AWS_SESSION_TOKEN=<awsSessionToken>
+
+.. note::
+   
+   Omit the line containing ``AWS_SESSION_TOKEN`` if you don't need an AWS
+   session token for that role.
+
+To authenticate by using **ECS container credentials**, set the ECS-endpoint-relative 
+URI in an environment variable by using ``bash``
+or a similar shell as shown in the following example:
+
+.. code-block:: bash
+
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<your ECS endpoint>
```

Comment:
AWS.SDK also supports this env variable AWS_CONTAINER_CREDENTIALS_FULL_URI

Reply:
Do you think we should mention both?

Reply:
Below you mention:
>To authenticate by using **EC2 container credentials**, make sure none of the environment variables mentioned earlier are set.

If you `AWS_CONTAINER_CREDENTIALS_FULL_URI`, the ec2 workflow will be skipped too. So I would say yes, better to mention it